### DISCUSSION TITLE
emacs mode

### DISCUSSION THREAD

**Discussion by andreltr** [google/or-tools#4450] [General]
Hello everyone,

Is there an emacs mode for or-tools developed by someone available?

TIA,

--
Andre Luiz Tietbohl Ramos, PhD

**Comment by lperron** [comment]
Why not use the python mode?

**Comment by andreltr** [comment]
Hello lperron,

Thanks for the response. Is or-tools' syntax coloring and indentation scheme compatible with python? It would be nice if so.

**Comment by lperron** [comment]
The are no languages in or-tools, only libraries.

**Comment by Mizux** [comment]
For C++ you have the cmake option:
https://github.com/google/or-tools/blob/d2badaa1f28ac24e086e09b80398929588f791c0/CMakeLists.txt#L18-L19
which should provide a `compile_commands.json` in the build dir, that you can symlink
`ln -s <build>/compile_commands.json .`.
Most completer/lsp can use it to provide C++ completion (helix, nvim, vscodium)...
